### PR TITLE
Omsi Hook RPC Plugin

### PR DESCRIPTION
**Pull request by space928**
This pull request isn't ready for merging quite yet and needs some extensive testing. Especially with regards to regression testing as I probably broke a bunch of stuff.

### CODE REVIEW

**Comment by winer222** [comment]
Looking great, ill run some tests this evening

File: OmsiHook/OmsiRemoteMethods.cs [hunk 1]
```diff
@@ -1,27 +1,53 @@
...
+            }
+#endif
+        }
 
         [Obsolete]
-        public int MakeVehicle()
+        public static int MakeVehicle()
         {
             int vehList = TTempRVListCreate(0x0074802C, 1);
             string path = @"Vehicles\GPM_MAN_LionsCity_M\MAN_A47.bus";
```
Comment (winer222) [inline OmsiHook/OmsiRemoteMethods.cs:46]: I'm guessing this is demo data - probs shouldnt be pushed to prod, maybe bring to an example code or something 🤷

File: OmsiHook/Memory.cs [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
+using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace OmsiHook
 {
     /// <summary>
-    /// Memory managment class for accessing and marshaling OMSI's memory
+    /// Memory management class for accessing and marshaling OMSI's memory
```
Comment (winer222) [inline OmsiHook/Memory.cs:14]: Pedantic :P

**Review by winer222 — APPROVED** [review]
Many changes, though none look too likely to break things outside of what you had already tested, im happy to approve - couple comments in there just to mess with you, please read them :P

File: OmsiHook/Memory.cs [hunk 3]
```diff
@@ -32,25 +31,28 @@ internal struct Flags
...
-                    Console.WriteLine("The plugin couldn't grant itself debug privileges, it may fail to attach to the game.");
-                    Console.WriteLine(e);
-                }
-                m_iProcessHandle =
-                    Imports.OpenProcess(Flags.PROCESS_VM_OPERATION | Flags.PROCESS_VM_READ | Flags.PROCESS_VM_WRITE,
-                        false, m_iProcess.Id);
-                return (true, m_iProcess);
+                Process.EnterDebugMode();
             }
+            catch (Exception e)
```
[the remark this answers is not in the document]
Comment (space928) [reply OmsiHook/Memory.cs:42]: `e` is used in Debug builds (see the preprocessor block below), but not in Release, I could fix this at the cost of adding another preprocessor block, but who cares...

File: OmsiHook/OmsiComplObjInst.cs
[the remark this answers is not in the document]
Comment (space928) [reply OmsiHook/OmsiComplObjInst.cs:1]: Yeah we should investigate this, I can't remember if any of these arrays worked correctly before (I know some were broken but I didn't think they all were). Might be a regression.

**Pull request merged**